lib/eunit/src/eunit_lib.erl

```diff
@@ -574,6 +574,17 @@ trie_match([], []) ->
 trie_match([], _T) ->
     no.
 
+%% ---------------------------------------------------------------------
+%% Convert a string of unknown encoding to a binary
+
+try_to_binary(S) when is_list(S) ->
```

Comment:
Lists can be both and it depends on the interpretation. ~s would expect a list of bytes (latin1) and ~ts would expect a list of code points and can go above 255. The best way to convert would probably just be to use `unicode:characters_to_binary/1-3` and let the library figure it out?